File: minimum-common-value.ts
{
    function getCommon(nums1: number[], nums2: number[]): number {
        let start = 0;
        let start_end = 0;
        
        while(nums1.length > start && nums2.length > start_end) {
            
            if(nums1[start] == nums2[start_end]) {
                return nums1[start]
            }

            if(nums1[start] < nums2[start_end]) {
                start++
            } else {
                start_end++
            }
            
        }

        return -1
    };  

    console.log(getCommon([3,5], [2]))

}

File: minimum-common-value.rs
pub fn get_common(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
    let (n1, n2) = (nums1.len(), nums2.len());
    let (mut i, mut j) = (0, 0);

    while i < n1 && j < n2 {
        if nums1[i] == nums2[j] {
            return nums1[i];
        }
        if nums1[i] < nums2[j] {
            i += 1;
        } else {
            j += 1;
        }
    }
    -1
}
